File: app/src/main/java/com/kieran/winnipegbusbackend/TripPlanner/SegmentFactory.java
package com.kieran.winnipegbusbackend.TripPlanner;

import com.kieran.winnipegbusbackend.TripPlanner.classes.RideSegment;
import com.kieran.winnipegbusbackend.TripPlanner.classes.Segment;
import com.kieran.winnipegbusbackend.TripPlanner.classes.TransferSegment;
import com.kieran.winnipegbusbackend.TripPlanner.classes.WalkSegment;

import org.json.JSONException;
import org.json.JSONObject;

public class SegmentFactory {
    public static Segment createSegment(JSONObject segment, String fromTitle, String toTitle) {
        try {
            switch (segment.getString("type")) {
                default: return null;
                case "walk": return new WalkSegment(segment, fromTitle, toTitle);
                case "ride": return new RideSegment(segment, fromTitle, toTitle);
                case "transfer": return new TransferSegment(segment, fromTitle, toTitle);
            }
        } catch (JSONException e) {
            return null;
        }
    }
}

File: app/src/main/java/com/kieran/winnipegbusbackend/TripPlanner/SegmentFactory.kt
package com.kieran.winnipegbusbackend.TripPlanner

import com.kieran.winnipegbusbackend.TripPlanner.classes.*

import org.json.JSONException
import org.json.JSONObject

object SegmentFactory {
    fun createSegment(tripParameters: TripParameters, segment: JSONObject): Segment? {
        return try {
            when (segment.getString("type")) {
                "walk" -> WalkSegment(tripParameters, segment)
                "ride" -> RideSegment(tripParameters, segment)
                "transfer" -> TransferSegment(tripParameters, segment)
                else -> null
            }
        } catch (e: JSONException) {
            null
        }

    }
}
